src/Sparrow.Server/Extensions/TcpExtensions.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Net.NetworkInformation;
+
+namespace Sparrow.Server.Extensions
+{
+    public static class TcpExtensions
+    {
+        public static IPGlobalProperties GetIPGlobalPropertiesSafely()
+        {
+            try
+            {
+                return IPGlobalProperties.GetIPGlobalProperties();
+            }
+            catch
+            {
+                return null;
```

Comment:
Should we return an empty object, instead?

Reply:
No, I'm taking that into account in other extension methods. Also all of the implementations are internal.